WcaOnRails/app/models/registration.rb

```diff
@@ -86,6 +83,10 @@ def best_solve(event, type)
     person ? person.best_solve(event, type) : SolveTime.new(event.id, type, 0)
   end
 
+  def event_picker_events
+    registration_events.map(&:event_object).sort_by(&:rank)
+  end
```

Comment:
@pingskills the main purpose of associated_events_picker was to keep the logic out of models (especially that all those models would have very similar code handling this). Doing something else for competition_events and registration_events seems messy as well. I've tried to search in google as @larspetrus suggested but haven't succeeded (maybe you have?). I've already commented with my propositions, thoughts?